persistent-test/src/PersistLiteralTestSQL.hs

```diff
@@ -0,0 +1,30 @@
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# OPTIONS_GHC -Wno-unused-top-binds #-}
+module PersistLiteralTestSQL (specsWith) where
+
+import Init
+import qualified Data.Text as T
+import NullableGenerated
+
+share [mkPersist sqlSettings { mpsGeneric = True }] [persistLowerCase|
+  PersistLiteralFieldTestTable
+    fieldOne Text
+    fieldTwo Text
+    fieldThree (NullableGenerated Text)
+    deriving Show Eq
+|]
+
+specsWith :: Runner SqlBackend m => RunDb SqlBackend m -> Spec
+specsWith runDB = describe "PersistLiteral field" $ do
+  it "should read a GENERATED column" $ runDB $ do
+    -- we have to manage the table manually because of the generated field
+    _ <- rawExecute "CREATE TABLE persist_literal_field_test_table (id serial primary key, field_one text, field_two text, field_three text GENERATED ALWAYS AS (COALESCE(field_one, field_two)) STORED);" []
```

Comment:
Possibly do a DROP TABLE IF EXISTS before this line, rather than a DROP TABLE after? If this test failed part way through, future runs of the test would fail in the current code, if I understand right

Reply:
For some reason I thought that `IF EXISTS` was not standard SQL. Just checked and it is at least supported by Postgres, MySQL, and SQLite, so I will implement this.